core/src/commonMain/kotlin/com/powersync/db/PowerSyncDatabaseImpl.kt

```diff
@@ -89,17 +97,22 @@ internal class PowerSyncDatabaseImpl(
     private var syncJob: Job? = null
     private var uploadJob: Job? = null
 
-    init {
-        val res = ActiveDatabaseGroup.referenceDatabase(logger, identifier)
-        resource = res.first
-        clearResourceWhenDisposed = res.second
+    // This is set in the init
+    private var powerSyncVersion: String? = null
```

Comment:
I suppose the typical pattern for this in Kotlin would be to

```suggestion
    private lateinit var powerSyncVersion: String
```

And then remove the null-checks when accessing it.

Reply:
That is nicer! I've implemented this.